src/NHibernate.Test/NHSpecificTest/NH2420/Fixture.cs

```diff
@@ -62,10 +62,8 @@ public void ShouldBeAbleToReleaseSuppliedConnectionAfterDistributedTransaction()
 						new DummyEnlistment(),
 						EnlistmentOptions.None);
 
-					if (Sfi.ConnectionProvider.Driver.GetType() == typeof(OdbcDriver))
-						connection = new OdbcConnection(connectionString);
-					else
-						connection = new SqlConnection(connectionString);
+					connection = Sfi.ConnectionProvider.Driver.CreateConnection();
+					connection.ConnectionString = connectionString;
```

Comment:
`Driver.CreateConnection()` should have been used from the start. It is meant for this. It just creates the connection, not doing anything else with it.